pkg/lib/infra/sms/hook.go

```diff
@@ -0,0 +1,23 @@
+package sms
+
+import (
+	"net/http"
+
+	"github.com/authgear/authgear-server/pkg/lib/config"
+	"github.com/authgear/authgear-server/pkg/util/httputil"
+)
+
+type HookHTTPClient struct {
+	*http.Client
+}
+
+func NewHookHTTPClient(hookCfg *config.HookConfig, smsCfg *config.CustomSMSProviderConfig) HookHTTPClient {
```

Comment:
In what case smsCfg is nil? We have `func (c *SecretConfig) Validate(appConfig *AppConfig) error` in pkg/lib/config/secret.go. We can enforce when sms provider is custom, the config is present.

I think we should not use the timeout of hook config as fallback. Use a static value as fallback.

Reply:
> In what case smsCfg is nil? We have func (c *SecretConfig) Validate(appConfig *AppConfig) error in pkg/lib/config/secret.go. We can enforce when sms provider is custom, the config is present.

When provider is not `custom`, `smsCfg` config will be nil, but in `NewCustomClient` if this config is nil, `CustomClient` will become nil therefore I found that actually I don't have to check it in `NewHookHTTPClient`.